### PR TITLE
non-latest tag should fail the validation

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related issues
fixes: https://github.com/demisto/etc/issues/25365

## Description
non-latest tag should fail the validation

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -421,6 +422,15 @@ def no_docker_tag(docker_image):
     def docker_not_formatted_correctly(docker_image):
         return f'The docker image: {docker_image} is not of format - demisto/image_name:X.X'
 
+    @staticmethod
+    @error_code_decorator
+    def docker_not_on_the_latest_tag(docker_image_tag, docker_image_latest_tag, docker_image_name):
+        return f'The docker image tag is not the latest, please update it.\n' \
```
Comment (ronykoz) [inline demisto_sdk/commands/common/errors.py:428]: please specify it should be the numeric value and not the tag latest

File: demisto_sdk/commands/common/hook_validations/description.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def is_duplicate_description(self):
         if not re.match(PACKS_INTEGRATION_YML_REGEX, self.file_path, re.IGNORECASE):
             package_path = os.path.dirname(self.file_path)
             try:
-                md_file_path = glob.glob(os.path.join(os.path.dirname(self.file_path), '*_description.md'))[0]
+                md_file_path = glob.glob(os.path.join(self.file_path.split('.')[0], '_description.md'))[0]
```
Comment (ronykoz) [inline demisto_sdk/commands/common/hook_validations/description.py:70]: why not 
os.path.basename(self.file_path)

File: demisto_sdk/commands/common/hook_validations/structure.py [hunk 3]
```diff
@@ -114,8 +114,9 @@ def is_valid_scheme(self):
             # disabling massages of level INFO and beneath of pykwalify such as: INFO:pykwalify.core:validation.valid
             log = logging.getLogger('pykwalify.core')
             log.setLevel(logging.WARNING)
+            scheme_file_name = 'integration' if self.scheme_name == 'betaintegration' else self.scheme_name
```
Comment (ronykoz) [inline demisto_sdk/commands/common/hook_validations/structure.py:117]: can't we point betaintegation to integration scheme ?

Comment (roysagi) [reply demisto_sdk/commands/common/hook_validations/structure.py:117]: this is exactly what is done here. we are pointing the beta to the integration scheme.

File: demisto_sdk/commands/common/errors.py (hunk 1 above)
Comment (roysagi) [reply demisto_sdk/commands/common/errors.py:428]: added, though the lines below tell what should be the changed tag.

File: demisto_sdk/commands/common/hook_validations/description.py (hunk 2 above)
Comment (roysagi) [reply demisto_sdk/commands/common/hook_validations/description.py:70]: I'm looking for the full path including file name without the extension  - I'll use: os.path.splitext instead

File: demisto_sdk/commands/validate/file_validator.py [hunk 4]
```diff
@@ -100,12 +100,16 @@ def __init__(self, is_backward_check=True, prev_ver=None, use_git=False, only_co
...
+        if self.validate_all:
+            self.skip_docker_checks = True
+
         self.branch_name = ''
         self.use_git = use_git
         self.skip_pack_rn_validation = skip_pack_rn_validation
         if self.use_git:
             print('Using git')
-            self.branch_name = self.get_current_working_branch()
+            self.branch_name = ''
```
Comment (ronykoz) [inline demisto_sdk/commands/validate/file_validator.py:112]: why do we need it then?

File: demisto_sdk/commands/validate/file_validator.py [hunk 5]
```diff
@@ -100,12 +100,16 @@ def __init__(self, is_backward_check=True, prev_ver=None, use_git=False, only_co
                  validate_all=False, is_private_repo=False, skip_pack_rn_validation=False, print_ignored_errors=False,
                  configuration=Configuration()):
         self.validate_all = validate_all
+        self.skip_docker_checks = False
+        if self.validate_all:
+            self.skip_docker_checks = True
```
Comment (ronykoz) [inline demisto_sdk/commands/validate/file_validator.py:105]:
```suggestion
        self.skip_docker_checks = True if self.validate_all else False
```

File: demisto_sdk/commands/validate/file_validator.py (hunk 4 above)
Comment (roysagi) [reply demisto_sdk/commands/validate/file_validator.py]: wrong change - fixing

**Pull request merged**